From 41bf6f3a7e974c6630a3f420d38f0b3f52984cfd Mon Sep 17 00:00:00 2001
From: Joshua Strobl <me@joshuastrobl.com>
Date: Sat, 3 Oct 2026 12:51:47 +0300
Subject: [PATCH] fix(session): dont force qpa wayland platform, fallback to
 xcb

Can prevent launching some older Qt apps, AppImages, etc.

Signed-off-by: Joshua Strobl <me@joshuastrobl.com>
---
 src/session/startbudgielabwc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/session/startbudgielabwc.in b/src/session/startbudgielabwc.in
index afdd542ee..be8bfaf83 100755
--- a/src/session/startbudgielabwc.in
+++ b/src/session/startbudgielabwc.in
@@ -67,7 +67,7 @@ if [ -z "${GDK_BACKEND}" ]; then
     export GDK_BACKEND
 fi
 if [ -z "${QT_QPA_PLATFORM}" ]; then
-    QT_QPA_PLATFORM="wayland"
+    QT_QPA_PLATFORM="wayland;xcb"
     export QT_QPA_PLATFORM
 fi
 
